eva/executor/function_scan_executor.py

```diff
@@ -0,0 +1,42 @@
+# coding=utf-8
+# Copyright 2018-2020 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Iterator
+
+from eva.planner.function_scan_plan import FunctionScanPlan
+from eva.executor.abstract_executor import AbstractExecutor
+from eva.models.storage.batch import Batch
+
+
+class FunctionScanExecutor(AbstractExecutor):
+    """
+    Executes functional expression which yields a table of rows
+    Arguments:
+        node (AbstractPlan): FunctionScanPlan
+
+    """
+
+    def __init__(self, node: FunctionScanPlan):
+        super().__init__(node)
+        self.func_expr = node.func_expr
+
+    def validate(self):
+        pass
+
+    def exec(self, lateral_input: Batch, *args, **kwargs) -> Iterator[Batch]:
```

Comment:
Consider the following plan:
```mermaid
graph TD;
  APPLY-->SELECTION;
  SELECTION-->FUNCTIONSCAN;
```
Now the Selection's executor also needs to pass `lateral_input`to its children's `exec()`. In other words, all operators between APPLY and FUNCTIONSCAN have to call their children's `exec()` differently. Can we make it anonymously to other operators? E.g., having `lateral_input` in `**kwargs` and passing `**kwargs` to children's `exec()` always.

Reply:
Sure, It should be that way